File: dsa/concepts/18_strings/java/L520_DetectCapital.java

public class L520_DetectCapital {

	public static void main(String[] args) {
		L520_DetectCapital obj = new L520_DetectCapital();

		String word = "USA";
		System.out.println(obj.detectCapitalUse(word));
	}

	/*
	 * Problem: https://leetcode.com/problems/detect-capital/
	 * Idea: Use startCapital and continuesCapital vars to track the status
     * TODO
	 * */
	public boolean detectCapitalUse(String word) {
		boolean startsCap = false, continuesCap = false;
        for(int i=0; i<word.length(); i++){
            char ch = word.charAt(i);
            if(i == 0)
                if(ch>='A' && ch<='Z') { startsCap = true; continue; }
            if(i == 1)
                if(ch>='A' && ch<='Z') continuesCap = true;

            if(startsCap && continuesCap && ch>='a' && ch<='z') return false;
            if(startsCap && !continuesCap && ch>='A' && ch<='Z') return false;
            if(!startsCap && ch>='A' && ch<='Z') return false;
        }

        return true;
    }

}

File: dsa/concepts/18_strings/python/l520_detect_capital.py
# Problem: https://leetcode.com/problems/detect-capital/
# Idea: Use startCapital and continuesCapital vars to track the status
# TODO
def detect_capital_use(word):
    starts_cap = False
    continues_cap = False

    for i in range(len(word)):
        ch = word[i]

        if i == 0:
            if 'A' <= ch <= 'Z':
                starts_cap = True
                continue

        if i == 1:
            if 'A' <= ch <= 'Z':
                continues_cap = True

        if starts_cap and continues_cap and 'a' <= ch <= 'z':
            return False
        if starts_cap and not continues_cap and 'A' <= ch <= 'Z':
            return False
        if not starts_cap and 'A' <= ch <= 'Z':
            return False

    return True


# ------------------ Test (Same as Java main) ------------------

word = "USA"
print(detect_capital_use(word))  # Expected: True
